src/astro/sql/operators/save_file.py

```diff
@@ -0,0 +1,86 @@
+from typing import Optional, Union
+
+import pandas as pd
+from airflow.models import BaseOperator
+from airflow.models.xcom_arg import XComArg
+
+from astro.constants import ExportExistsStrategy
+from astro.databases import create_database
+from astro.files import File
+from astro.sql.table import Table
+from astro.utils.task_id_helper import get_task_id
+
+
+class SaveFile(BaseOperator):
+    """Write SQL table to csv/parquet on local/S3/GCS.
+
+    :param input_data: Table to convert to file
+    :param output_file: File object containing the path to the file and connection id.
+    :param if_exists: Overwrite file if exists. Default False.
+    """
+
+    template_fields = ("input_data", "output_file")
+
+    def __init__(
+        self,
+        input_data: Union[Table, pd.DataFrame],
+        output_file: File,
+        if_exists: ExportExistsStrategy = "exception",
+        **kwargs,
+    ) -> None:
+        super().__init__(**kwargs)
+        self.output_file = output_file
+        self.input_data = input_data
+        self.if_exists = if_exists
+        self.kwargs = kwargs
+
+    def execute(self, context: dict) -> None:
+        """Write SQL table to csv/parquet on local/S3/GCS.
+
+        Infers SQL database type based on connection.
+        """
+        # Infer db type from `input_conn_id`.
+        if isinstance(self.input_data, Table):
+            database = create_database(self.input_data.conn_id)
```

Comment:
Before fetching the data from the source table, for consistency, we should use the Database.populate_table_metadata:
https://github.com/astronomer/astro-sdk/blob/main/src/astro/databases/base.py#L114
Similar to:
https://github.com/astronomer/astro-sdk/blob/main/src/astro/sql/operators/transform.py#L267

So we retrieve metadata from the connection or environment variable if needed.

Reply:
Done.